src/pkg/utils/oci.go

```diff
@@ -52,23 +53,24 @@ func ToOCIStore(t any, mediaType string, store *ocistore.Store) (ocispec.Descrip
 }
 
 // ToOCIRemote takes an arbitrary type, typically a struct, marshals it into JSON and store it in a remote OCI store
-func ToOCIRemote(t any, mediaType string, remote *oci.OrasRemote) (ocispec.Descriptor, error) {
+func ToOCIRemote(ctx context.Context, t any, mediaType string, remote *oci.OrasRemote) (*ocispec.Descriptor, error) {
 	b, err := json.Marshal(t)
 	if err != nil {
-		return ocispec.Descriptor{}, err
+		return &ocispec.Descriptor{}, err
 	}
 
-	var layerDesc ocispec.Descriptor
+	var layerDesc *ocispec.Descriptor
 	// if image manifest media type, push to Manifests(), otherwise normal pushLayer()
 	if mediaType == ocispec.MediaTypeImageManifest {
-		layerDesc = content.NewDescriptorFromBytes(ocispec.MediaTypeImageManifest, b)
-		if err := remote.Repo().Manifests().PushReference(context.TODO(), layerDesc, bytes.NewReader(b), remote.Repo().Reference.String()); err != nil {
-			return ocispec.Descriptor{}, fmt.Errorf("failed to push manifest: %w", err)
+		descriptorFromBytes := content.NewDescriptorFromBytes(ocispec.MediaTypeImageManifest, b)
+		layerDesc = &descriptorFromBytes
```

Comment:
I think you can kill `layerDesc = &descriptorFromBytes` and just pass in `descriptorFromBytes ` to `PushReference()`

Reply:
so the layerDesc ends up getting used at the end of the function as part of the success message regardless of which if block you end up in.